extensions/lifecycle/CircuitBreakerGroup.js

```diff
@@ -0,0 +1,317 @@
+ * Invalid probes are ignored
+ * @param {Object} cbConf templated circuit breaker configuration
+ * @param {Object} lcConfig lifecycle configuration
+ * @param {Object} repConfig replication configuration
+ * @param {Object} locations location configuration
+ * @param {Logger} logger logger instance
+ * @returns {BucketProcessorCircuitBreakerConfig} circuit breaker configuration
+ */
+function extractBucketProcessorCircuitBreakerConfigs(cbConf, lcConfig, repConfig, locations, logger) {
+    if (!cbConf || !cbConf.probes) {
+        return {
+            global: {},
+            circuitBreakerGroup: new CircuitBreakerGroup({}, null),
+        };
+    }
+
+    const { probes, ...globalCircuitBreakerConf } = cbConf;
+
+    const circuitBreakerGroup = new CircuitBreakerGroup(globalCircuitBreakerConf, null);
+
+    // config off global circuit breaker
+    const global = {
+        ...globalCircuitBreakerConf,
+        probes: [],
+    };
+
+    probes.forEach(probe => {
+        if (probe.type !== 'prometheusQuery') {
+            global.probes.push(probe);
+            return;
+        }
+        // when clause is optional and can be used to specify the filtering on
+        // the topic and location, along with specifying if a probe should be
+        // used for transition and expiration.
+        // Query syntax:
+        // "when{topic=topicName,location=locationName,transition=true,expiration=true} and ANY"
+        // Note: should either specify the topic or the location, not both
+        // Behaviour:
+        // - no when clause + templated topic: creates a circuit breaker for each topic
+        // - no when clause + templated location: creates a circuit breaker for each location
+        // - when clause with a specific topic + templated topic: creates a circuit breaker for the specified topic
+        // - when clause with a specific location + templated location: creates a circuit breaker for the location
+        // - when clause with topic and location + topic and location template: creates a circuit breaker for the
+        // location and topic
+        // - topic specified + location template : creates a circuit breaker for each location and references them in
+        // the topic
+        // - location specified + topic template : creates a circuit breaker for each topic and references them in
+        // the location
+        const whenClause = probe.query.match(/^when\{(.*?)\}\sand\s/);
+
+        const withTopicTemplate = probe.query.includes('${topic}');
+        const withLocationTemplate = probe.query.includes('${location}');
+
+        const topic = whenClause && whenClause[1].match(/topic=(.*?)(,|$)/);
+        const location = whenClause && whenClause[1].match(/location=(.*?)(,|$)/);
+
+        const transition = whenClause && whenClause[1].match(/transition=(.*?)(,|$)/);
+        const expiration = whenClause && whenClause[1].match(/expiration=(.*?)(,|$)/);
```

Comment:
should be quoted here as well: prometheus does not support `bool` literals
```suggestion
        const transition = whenClause && whenClause[1].match(/transition="(.*?)"(,|$)/);
        const expiration = whenClause && whenClause[1].match(/expiration="(.*?)"(,|$)/);
```

maybe we can also be more strict?
 ```suggestion
        const transition = whenClause && whenClause[1].match(/transition="(true|false)"(,|$)/);
        const expiration = whenClause && whenClause[1].match(/expiration="(true|false)"(,|$)/);
```